avm/ptn/data/private-analytical-workspace/main.bicep

```diff
@@ -0,0 +1,864 @@
+    tags: tags
+    virtualNetworkLinks: [
+      {
+        registrationEnabled: false
+        virtualNetworkResourceId: vnet.outputs.resourceId
+      }
+    ]
+  }
+}
+
+// ============ //
+// Outputs      //
+// ============ //
+
+@description('The resource ID of the resource.')
+output resourceId string = vnetCfg.resourceId
+
+@description('The name of the resource.')
+output name string = vnetCfg.name
+
+@description('The location the resource was deployed into.')
+output location string = vnetCfg.location
+
+@description('The name of the managed resource group.')
+output resourceGroupName string = vnetCfg.resourceGroupName
+
+@description('The resource ID of the Azure Virtual Network.')
+output virtualNetworkResourceId string = vnetCfg.resourceId
+
+@description('The name of the Azure Virtual Network.')
+output virtualNetworkName string = vnetCfg.name
+
+@description('The location of the Azure Virtual Network.')
+output virtualNetworkLocation string = vnetCfg.location
+
+@description('The name of the Azure Virtual Network resource group.')
+output virtualNetworkResourceGroupName string = vnetCfg.resourceGroupName
+
+@description('The resource ID of the Azure Log Analytics Workspace.')
+output logAnalyticsWorkspaceResourceId string = logCfg.resourceId
+
+@description('The name of the Azure Log Analytics Workspace.')
+output logAnalyticsWorkspaceName string = logCfg.name
+
+@description('The location of the Azure Log Analytics Workspace.')
+output logAnalyticsWorkspaceLocation string = logCfg.location
+
+@description('The name of the Azure Log Analytics Workspace resource group.')
+output logAnalyticsWorkspaceResourceGroupName string = logCfg.resourceGroupName
+
+@description('The resource ID of the Azure Key Vault.')
+output keyVaultResourceId string = kvCfg.resourceId
+
+@description('The name of the Azure Key Vault.')
+output keyVaultName string = kvCfg.name
+
+@description('The location of the Azure Key Vault.')
+output keyVaultLocation string = kvCfg.location
+
+@description('The name of the Azure Key Vault resource group.')
+output keyVaultResourceGroupName string = kvCfg.resourceGroupName
+
+@description('Conditional. The resource ID of the Azure Databricks when `enableDatabricks` is `true`.')
+output databricksResourceId string = enableDatabricks ? dbw.outputs.resourceId : ''
+
+@description('Conditional. The name of the Azure Databricks when `enableDatabricks` is `true`.')
+output databricksName string = enableDatabricks ? dbw.outputs.name : ''
+
+@description('Conditional. The location of the Azure Databricks when `enableDatabricks` is `true`.')
+output databricksLocation string = enableDatabricks ? dbw.outputs.location : ''
+
+@description('Conditional. The name of the Azure Databricks resource group when `enableDatabricks` is `true`.')
+output databricksResourceGroupName string = enableDatabricks ? dbw.outputs.resourceGroupName : ''
+
+// ================ //
+// Definitions      //
+// ================ //
+
+type lockType = {
```

Comment:
We'll soon at a spec to ask contributors to add the `@export()` tag to their UDTs - if - they should not remain private. This has the perk that consumers of your module can import those types in their templates when using your module to re-use them. 
If you don't mind, I'd ask you to please add this small annotation to each of your types. There is no impact on your module's functionality.

Reply:
Implemented propose changes